packages/design-system/src/components/Drawer/usePrevious.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef } from 'react';
+
+// storing a previous version of a prop for comparison
+// similar to the old previousProps param from `componentDidUpdate`
+const usePrevious = <T>(value: T): T | undefined => {
+  const ref = useRef<T>();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export default usePrevious;
```

Comment:
I wonder if this custom hook should be moved to a new `/hooks` directory - it might be useful if we want to use it in any other components that need some kind of `componentdidupdate(prevprops)` equivalent.